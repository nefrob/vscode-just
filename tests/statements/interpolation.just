# Interpolation blocks

# TODO: interpolation blocks don't really work highlighting-wise

base:
  echo {{var}} {{ var }}
  echo {{ a + 'b' + c }}
  echo {{ 'a' / 'b' }}
  echo '{{ var }}'
  echo '''{{ var }}'''
  echo '{{ 'var' }}'
  echo '''{{ 'var' }}'''

# Braces are escaped

escaped:
  echo '{{{{ var }}'
  echo '''{{{{ var }}'''
  echo {{{{ var }}
  echo '{{ "{{" }} var }}'
  echo {{ "{{" }} var }}
  echo '{{'{{ var }}'}}'
  # TODO: broken
  # echo {{ '{{ var }}' }}
  # echo {{ "{{ var" }} }}
  # echo {{ "{{ 'var' }}" }}
