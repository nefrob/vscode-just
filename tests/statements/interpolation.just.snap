># Interpolation blocks
#^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
># TODO: interpolation blocks don't really work highlighting-wise
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>base:
#^^^^ source.just entity.name.function.just
#    ^ source.just keyword.operator.recipe.end.just
>  echo {{var}} {{ var }}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^^^ source.just string.interpolated.escaping.just
#            ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#              ^ source.just
#               ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                 ^^^^^ source.just string.interpolated.escaping.just
#                      ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  echo {{ a + 'b' + c }}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^^^ source.just string.interpolated.escaping.just
#            ^ source.just string.interpolated.escaping.just keyword.operator.concat.just
#             ^ source.just string.interpolated.escaping.just
#              ^ source.just string.interpolated.escaping.just string.quoted.single.just string.quoted.single.just
#               ^ source.just string.interpolated.escaping.just string.quoted.single.just
#                ^ source.just string.interpolated.escaping.just string.quoted.single.just
#                 ^ source.just string.interpolated.escaping.just
#                  ^ source.just string.interpolated.escaping.just keyword.operator.concat.just
#                   ^^^ source.just string.interpolated.escaping.just
#                      ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  echo {{ 'a' / 'b' }}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^ source.just string.interpolated.escaping.just
#          ^ source.just string.interpolated.escaping.just string.quoted.single.just string.quoted.single.just
#           ^ source.just string.interpolated.escaping.just string.quoted.single.just
#            ^ source.just string.interpolated.escaping.just string.quoted.single.just
#             ^ source.just string.interpolated.escaping.just
#              ^ source.just string.interpolated.escaping.just keyword.operator.path-join.just
#               ^ source.just string.interpolated.escaping.just
#                ^ source.just string.interpolated.escaping.just string.quoted.single.just string.quoted.single.just
#                 ^ source.just string.interpolated.escaping.just string.quoted.single.just
#                  ^ source.just string.interpolated.escaping.just string.quoted.single.just
#                   ^ source.just string.interpolated.escaping.just
#                    ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  echo '{{ var }}'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#          ^^^^^ source.just string.quoted.single.just string.interpolated.escaping.just
#               ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                 ^ source.just string.quoted.single.just
>  echo '''{{ var }}'''
#^^^^^^^ source.just
#       ^^^ source.just string.quoted.single.indented.just string.quoted.single.indented.just
#          ^^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.interpolated.escape.just
#            ^^^^^ source.just string.quoted.single.indented.just string.interpolated.escaping.just
#                 ^^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.interpolated.escape.just
#                   ^^^ source.just string.quoted.single.indented.just
>  echo '{{ 'var' }}'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#          ^ source.just string.quoted.single.just string.interpolated.escaping.just
#           ^ source.just string.quoted.single.just string.interpolated.escaping.just string.quoted.single.just string.quoted.single.just
#            ^^^ source.just string.quoted.single.just string.interpolated.escaping.just string.quoted.single.just
#               ^ source.just string.quoted.single.just string.interpolated.escaping.just string.quoted.single.just
#                ^ source.just string.quoted.single.just string.interpolated.escaping.just
#                 ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                   ^ source.just string.quoted.single.just
>  echo '''{{ 'var' }}'''
#^^^^^^^ source.just
#       ^^^ source.just string.quoted.single.indented.just string.quoted.single.indented.just
#          ^^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.interpolated.escape.just
#            ^ source.just string.quoted.single.indented.just string.interpolated.escaping.just
#             ^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.quoted.single.just string.quoted.single.just
#              ^^^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.quoted.single.just
#                 ^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.quoted.single.just
#                  ^ source.just string.quoted.single.indented.just string.interpolated.escaping.just
#                   ^^ source.just string.quoted.single.indented.just string.interpolated.escaping.just string.interpolated.escape.just
#                     ^^^ source.just string.quoted.single.indented.just
>
># Braces are escaped
#^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>escaped:
#^^^^^^^ source.just entity.name.function.just
#       ^ source.just keyword.operator.recipe.end.just
>  echo '{{{{ var }}'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^^^^^^^ source.just string.quoted.single.just
#                   ^ source.just string.quoted.single.just
>  echo '''{{{{ var }}'''
#^^^^^^^ source.just
#       ^^^ source.just string.quoted.single.indented.just string.quoted.single.indented.just
#          ^^^^^^^^^^^ source.just string.quoted.single.indented.just
#                     ^^^ source.just string.quoted.single.indented.just
>  echo {{{{ var }}
#^^^^^^^^^^^^^^^^^^^ source.just
>  echo '{{ "{{" }} var }}'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#          ^ source.just string.quoted.single.just string.interpolated.escaping.just
#           ^^^^ source.just string.quoted.single.just string.interpolated.escaping.just string.quoted.double.indented.just
#               ^ source.just string.quoted.single.just string.interpolated.escaping.just
#                ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                  ^^^^^^^ source.just string.quoted.single.just
#                         ^ source.just string.quoted.single.just
>  echo {{ "{{" }} var }}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^ source.just string.interpolated.escaping.just
#          ^^^^ source.just string.interpolated.escaping.just string.quoted.double.indented.just
#              ^ source.just string.interpolated.escaping.just
#               ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                 ^^^^^^^^ source.just
>  echo '{{'{{ var }}'}}'
#^^^^^^^ source.just
#       ^^^^ source.just string.quoted.double.indented.just
#           ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#             ^^^^^ source.just string.interpolated.escaping.just
#                  ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                    ^ source.just string.quoted.single.just string.quoted.single.just
#                     ^^ source.just string.quoted.single.just
#                       ^ source.just string.quoted.single.just
>  # TODO: broken
#^^ source.just
#  ^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>  # echo {{ '{{ var }}' }}
#^^ source.just
#  ^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>  # echo {{ "{{ var" }} }}
#^^ source.just
#  ^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>  # echo {{ "{{ 'var' }}" }}
#^^ source.just
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>